OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs

```diff
@@ -298,36 +301,37 @@ class MoveDeployOrderTargeter : IOrderTargeter
 		{
 			readonly MobileInfo mobile;
 			readonly LocomotorInfo locomotorInfo;
-			readonly bool rejectMove;
 			readonly GrantConditionOnDeploy unit;
 
 			public bool TargetOverridesSelection(TargetModifiers modifiers)
 			{
 				return modifiers.HasModifier(TargetModifiers.ForceMove);
 			}
 
-			public MoveDeployOrderTargeter(Actor self, GrantConditionOnDeploy unit)
+			public MoveDeployOrderTargeter(ActorInfo actorInfo, GrantConditionOnDeploy unit)
 			{
 				this.unit = unit;
-				mobile = self.TraitOrDefault<Mobile>().Info;
+				mobile = actorInfo.TraitInfoOrDefault<MobileInfo>();
 				locomotorInfo = mobile.LocomotorInfo;
-				rejectMove = !self.AcceptsOrder("Move");
```

Comment:
We need to be careful about this, there are cases in missions where we want actors to be able to move in response to map scripts, have their weapons controlled by the player, but reject any player given move orders. This was implemented via the `RejectsOrders` trait and this check.

Reply:
`rejectMove` is put back in.